ibis/expr/analysis.py

```diff
@@ -30,29 +19,39 @@
 
 
 def sub_for(expr, substitutions):
-    mapping = {repr(k.op()): v for k, v in substitutions}
-    substitutor = Substitutor()
-    return substitutor.substitute(expr, mapping)
+    """Substitute subexpressions in `expr` with expression to expression
+    mapping `substitutions`.
```

Comment:
I got sweaty

Reply:
Yes, now that you've pointed it out it's making my head hurt.